libraries/cms/session/CsrfToken.php

```diff
@@ -0,0 +1,178 @@
+<?php
+/**
+ * @package     Joomla.Libraries
+ * @subpackage  Session
+ *
+ * @copyright   Copyright (C) 2005 - 2015 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE
+ */
+
+namespace Joomla\Cms\Session;
+
+defined('JPATH_PLATFORM') or die;
+
+/**
+ * Class CsrfToken
+ *
+ * @package  Joomla\Cms\Session
+ *
+ * @since    4.0
+ */
+class CsrfToken implements CsrfTokenInterface
+{
+	/** @var \JSession The session object */
+	private $session;
+
+	/**
+	 * Constructor
+	 *
+	 * @param   \JSession  $session  The session object
+	 */
+	public function __construct(\JSession $session)
+	{
+		$this->session = $session;
+	}
+
+	/**
+	 * Get a session token, if a token isn't set yet one will be generated.
+	 *
+	 * Tokens are used to secure forms from spamming attacks. Once a token
+	 * has been generated the system will check the post request to see if
```

Comment:
This docblock is misleading. The system doesn't check the POST request automatically. You have to use has() yourself. It should also be pointed out that the check() and guard() methods do not take into account the session token but the static form token that only depends on the site and the logged in user (not an anti-spam method – once you read it on a site you can submit an unlimited number of forms as the same user, from the same or a different session).